### DISCUSSION TITLE
Is there a way to use variables to create fields in a loop?

### DISCUSSION THREAD

**Discussion by cds-internationalist** [typst/typst#1156] [Q&A] [answered]
As a simple example, I'm trying to write something along the lines of;

`#let name_array = ("Steve","Maria","Ahmed")`
`#let participant_result_dict = (:)`
`#for name in name_array {`
 `participant_result_dict.name = " "`
`}`

In this case, the idea is to create a dictionary with keys matching the given array. The above code runs, but would only produce a single field named "name". Is there a way to use the content of the variable name as the name of a key being created in the loop?

**Comment by PgBiel** [comment, marked as answer]
Use `dict.insert(name, " ")`

**Reply by cds-internationalist** [reply]
Of course! Thank you.
